src/connectors/IConnector.ts

```diff
@@ -48,6 +48,10 @@ export type ConnectorError =
      * Represents a SIP handshake error.
      */
     | { type: 'ConnectorError'; name: 'SIP failed'; detail: string }
+    /**
+     * Represents access rejected error.
+     */
+    | { type: 'ConnectorError'; name: 'Access rejected'; detail?: string }
```

Comment:
No need to add detail optional if not need.
All the error are individual, if this one has no detail need, don't put it to avoid to reuse the same error to one other usage or location. The worse case to avoid for example:
`{ type: 'ConnectorError'; name: string ; detail?: string }`
Exactly what we want to avoid